ios/Classes/DocumentViewFactory.m

```diff
@@ -0,0 +1,28 @@
+#import "FlutterDocumentView.h"
+#import "PdftronFlutterPlugin.h"
+#import "PTPluginUtils.h"
+
+@implementation DocumentViewFactory {
+    NSObject<FlutterBinaryMessenger>* _messenger;
+}
+
+- (instancetype)initWithMessenger:(NSObject<FlutterBinaryMessenger> *)messenger
+{
+    self = [super init];
+    if (self) {
+        _messenger = messenger;
+    }
+    return self;
+}
+
+- (NSObject<FlutterMessageCodec> *)createArgsCodec
+{
+    return [FlutterStandardMessageCodec sharedInstance];
+}
+
+- (NSObject<FlutterPlatformView> *)createWithFrame:(CGRect)frame viewIdentifier:(int64_t)viewId arguments:(id)args
+{
+    return [PdftronFlutterPlugin registerWithFrame:frame viewIdentifier:viewId messenger:_messenger];
```

Comment:
You should use `self.messenger`, not `_messenger`. Generally only use the backing variable (`_messenger`) directly in the initializer (where it is done this way because otherwise you can get into trouble when using inheritance.)